DataFormats/Detectors/TRD/include/DataFormatsTRD/RawData.h

```diff
@@ -276,160 +279,25 @@ uint32_t getSlope(const *rawdata, int tracklet)
 */
 
-uint32_t unpacklinkinfo(const HalfCRUHeader& cruhead, const uint32_t link, const bool data = true)
-{
-  //TODO this needs some more ellaborate testing. I got the packing to work so for now,
-  //     its a question of unpacking info in line 88 and 89.
-  uint32_t info = 0;
-  if (link > 14)
-    return 0xffffffff;
-
-  char* words;
-  if (link < 8)
-    words = (char*)&cruhead.word02[0];
-  if (link < 15)
-    words = (char*)&cruhead.word57[0];
-  uint32_t byteoffset = link * 3; // each link [size+erroflags] is 24 bits, so 3 bytes.
-  if (data)
-    info = ((words[byteoffset + 1]) << 8) + (words[byteoffset + 2]);
-  else
-    info = words[byteoffset];
-
-  return info;
-}
-
-uint32_t getlinkerrorflag(const HalfCRUHeader& cruhead, const uint32_t link)
-{
-  uint32_t errorflags = 0;
-  if (link < 8) {
-    //dealing with word0-2
-    errorflags = cruhead.linksA[link].errorflags;
-  } else {
-    if (link < 16) {
-      errorflags = cruhead.linksB[link - 8 + 1].errorflags; // link 0 [actually 9] is in fact the end part of reserved.
-
-    } else
-      std::cout << "error link=" << link << " not in range 0-14" << std::endl;
-  }
-  return errorflags;
-}
-
-uint32_t getlinkdatasize(const HalfCRUHeader& cruhead, const uint32_t link)
-{
-  //return number 32 byte blocks for the link 3x64bit ints.
-  uint32_t size = 0;
-  if (link < 8) {
-    size = (cruhead.linksA[link].size);
-  } else {
-    if (link < 16) { // link 0 is part of reserved
-      size = cruhead.linksB[link - 8 + 1].size;
-
-    } else
-      std::cout << "error link=" << link << " not in range 0-14" << std::endl;
-  }
-  return size;
-}
-
-uint32_t getlinkerrorflags(const HalfCRUHeader& cruheader, std::array<uint32_t, 15>& linkerrorflags)
-{
-  for (uint32_t link = 0; link < 15; link++) {
-    linkerrorflags[link] = getlinkerrorflag(cruheader, link);
-  }
-  return 0;
-}
-uint32_t getlinkdatasizes(const HalfCRUHeader& cruheader, std::array<uint32_t, 15>& linksizes)
-{
-  for (uint32_t link = 0; link < 15; link++) {
-    linksizes[link] = getlinkdatasize(cruheader, link);
-  }
-  return 0;
-}
-
-/*
- *  Printing methods to dump and display the various structures above in pretty format or hexdump
- */
-
-std::ostream& operator<<(std::ostream& stream, const HalfChamberHeader halfchamberheader)
-{
-  stream << "HalfChamberHeader : " << halfchamberheader.format << " ;; " << halfchamberheader.MCLK << " :: " << halfchamberheader.one << " :: " << halfchamberheader.HCID << std::endl;
-  return stream;
-}
-
-std::ostream& operator<<(std::ostream& stream, const TrapRawTracklet& tracklet)
-{
-  // make a pretty output of the tracklet.
-  stream << "TrapRawTracklet: pos=" << tracklet.pos << "::slope=" << tracklet.slope << "::pid=" << tracklet.pid << "::checkbit=" << tracklet.checkbit << std::endl;
-  return stream;
-}
-void printTrapRawTracklet(o2::trd::TrapRawTracklet& tracklet)
-{
-  LOGF(INFO, "TrapRawTracklet: ");
-}
 
-void printMCMHeader(o2::trd::MCMRawDataHeader& mcmhead)
```

Comment:
I just spotted an issue here: you are actually printing a tracklet header. The name MCMRawDataHeader would more intuitively be reserved for the digit/ADC data headers.
Should we change this now, or do a review of the raw data struct in the next week or two?

Reply:
Now would be the best time to change it yes. We can discuss in the current meeting.